tests/msc2716_test.go

```diff
@@ -106,15 +112,15 @@ func TestBackfillingHistory(t *testing.T) {
 			eventIDsAfter := createMessagesInRoom(t, alice, roomID, 2)
 
 			// Insert the most recent chunk of backfilled history
+			insertTime1 := timeAfterEventBefore.Add(timeBetweenMessages * 3)
 			batchSendRes := batchSendHistoricalMessages(
 				t,
 				as,
-				[]string{virtualUserID},
 				roomID,
 				eventIdBefore,
-				timeAfterEventBefore.Add(timeBetweenMessages*3),
 				"",
-				3,
+				createJoinStateEventsForBackfillRequest([]string{virtualUserID}, insertTime1),
+				createMessageEventsForBackfillRequest([]string{virtualUserID}, insertTime1, 3),
```

Comment:
Simplified the function signatures a bit by making these composable functions which have a nice name to indicate what they are doing.

Spurred on by the new private room test which also needs invite state events added. So now you can mix and match your favorite event generation to get what you want for the test.